CMakeLists.txt

```diff
@@ -641,7 +641,6 @@ if(IREE_BUILD_PYTHON_BINDINGS)
   # seems to be robust generally.
   # See: https://reviews.llvm.org/D118148
   # If building Python packages, we have a hard requirement on 3.9+.
-  find_package(Python3 3.9 COMPONENTS Interpreter Development NumPy)
```

Comment:
If this is truly safe to remove now, then also reword the comments above.

This in particular is explaining why there are two similar calls to `find_package` back to back: 
```
  # Configuring the Development.Module is flaky in multi-project setups.
  # "Bootstrapping" by first looking for the optional Development component
  # seems to be robust generally.
  # See: https://reviews.llvm.org/D118148
```